org.idempiere.test/src/org/idempiere/test/acct/AccountingSchemaValidRangeTest.java

```diff
@@ -0,0 +1,263 @@
+											getTrxName());
+
+			// Step 3: Bank Account Acct - Clone for new schema
+			sql = new StringBuilder("");
+			sql	.append("INSERT INTO C_BankAccount_Acct 												")
+				.append("	(	C_BankAccount_ID, C_AcctSchema_ID, 										")
+				.append(" 		AD_Client_ID, AD_Org_ID, IsActive, Created,CreatedBy, Updated,UpdatedBy,")
+				.append(" 		B_Asset_Acct, B_InTransit_Acct, B_PaymentSelect_Acct, 					")
+				.append(" 		B_UnallocatedCash_Acct, B_InterestExp_Acct, B_InterestRev_Acct 		) 	")
+				.append("	SELECT C_BankAccount_ID, ").append(acctSchema.get_ID()).append(", 			")
+				.append(" 		AD_Client_ID, AD_Org_ID, 'Y', getDate(), 0, getDate(), 0,				")
+				.append(" 		B_Asset_Acct, B_InTransit_Acct, B_PaymentSelect_Acct,					")
+				.append("		B_UnallocatedCash_Acct, B_InterestExp_Acct, B_InterestRev_Acct 			")
+				.append(" FROM C_BankAccount_Acct ")
+				.append(" WHERE C_AcctSchema_ID = ").append(DictionaryIDs.C_AcctSchema.DOLLAR.id)
+				.append(" 	AND C_BankAccount_ID = ").append(DictionaryIDs.C_BankAccount.ACCOUNT_1234.id);
+			DB.executeUpdate(sql.toString(), getTrxName());
+			if (DB.isGenerateUUIDSupported())
+				DB.executeUpdateEx("UPDATE C_BankAccount_Acct SET C_BankAccount_Acct_UU=generate_uuid() WHERE C_BankAccount_Acct_UU IS NULL", getTrxName());
+			else
+				UUIDGenerator.updateUUID(	MColumn.get(Env.getCtx(), X_C_BankAccount_Acct.Table_Name, PO.getUUIDColumnName(X_C_BankAccount_Acct.Table_Name)),
+											getTrxName());
+
+			//
+			CacheMgt.get().reset();
+
+			// Step 4: Create Payments and post
+
+			/***
+			 * DocType Is Always Post as True
+			 */
+
+			// Payment 1: Inside schema date range
+			List<MFactAcct> factAccts = null;
+			factAccts = docCompleteAndPostWithFactEntries(acctSchema, bp, docType, pmt_DateInRange, 111);
+			assertTrue(factAccts.size() > 0, "Posting should exist for payment in date range & DocType is always post as True");
+
+			// Payment 2: Outside schema date range
+			factAccts = docCompleteAndPostWithFactEntries(acctSchema, bp, docType, pmt_DateOutOfRange, 333);
+			assertTrue(factAccts.size() > 0, "Posting should exist for payment outside schema date range & DocType is always post as True");
+
+			/***
+			 * DocType Is Always Post as False
+			 */
+			docType.setIsAlwaysPosted(false);
+			docType.saveEx(null);
+			docType.load(null);
+
+			// Payment 3: Inside schema date range
+			factAccts = docCompleteAndPostWithFactEntries(acctSchema, bp, docType, pmt_DateInRange, 555);
+			assertTrue(factAccts.size() > 0, "Posting should exist for payment in date range with DocType is always post as False");
+
+			// Payment 4: Outside schema date range
+			factAccts = docCompleteAndPostWithFactEntries(acctSchema, bp, docType, pmt_DateOutOfRange, 777);
+			assertTrue(factAccts.size() == 0, "Posting should exist for payment outside schema date range with DocType is always post as False");
```

Comment:
I guess the message should be "Posting should not exist" instead ?

Reply:
You are correct, Just pushed correction